companion/gateway/routers/personality.py

```
@@ -0,0 +1,438 @@
+        raise HTTPException(status_code=500, detail=f"Personality engine error: {str(e)}")
+    except Exception as e:
+        logger.error(f"Error reinforcing quirk '{quirk_name}' for {user_id}: {e}")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@router.post("/need/update/{user_id}/{need_type}", response_model=PsychologicalNeed)
+async def update_need_level(
+    user_id: str,
+    need_type: str,
+    new_level: float = Query(..., ge=0.0, le=1.0),
+    personality_engine: PersonalityEngine = Depends(get_personality),
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)